roles/manage_machine_image/tasks/create.yml

```diff
@@ -0,0 +1,140 @@
+              - name = {{ manage_machine_image_source_disk_name }}
+          register: manage_machine_image__source_disk_info
+
+        - name: Print debug message
+          ansible.builtin.debug:
+            msg: Disk {{ manage_machine_image_source_source_disk_name }} does not exist.
+          when: manage_machine_image__source_disk_info.resources | length == 0
+
+    - name: Verify if snapshot exists
+      when: manage_machine_image_source_snapshot_name is defined
+      block:
+        - name: Get information about the snapshot {{ manage_machine_image_source_snapshot_name }}
+          google.cloud.gcp_compute_snapshot_info:
+            filters:
+              - name = {{ manage_machine_image_source_snapshot_name }}
+          register: manage_machine_image__source_snapshot_info
+
+        - name: Print debug message
+          ansible.builtin.debug:
+            msg: Snapshot {{ manage_machine_image_source_snapshot_name }} does not exist.
+          when: manage_machine_image__source_snapshot_info.resources | length == 0
+
+    - name: Verify if image exists
+      when: manage_machine_image_source_image_name is defined
+      block:
+        - name: Get information about the image {{ manage_machine_image_source_image_name }}
+          google.cloud.gcp_compute_image_info:
+            filters:
+              - name = {{ manage_machine_image_source_image_name }}
+          register: manage_machine_image__source_image_info
+
+        - name: Print debug message
+          ansible.builtin.debug:
+            msg: Image {{ manage_machine_image_source_image_name }} does not exist.
+          when: manage_machine_image__source_image_info.resources | length == 0
+
+    - name: Set 'manage_machine_image__source'
+      ansible.builtin.set_fact:
+        manage_machine_image__source: "{{ [manage_machine_image__source_disk_info.resources, manage_machine_image__source_snapshot_info.resources, manage_machine_image__source_image_info.resources] | reject('undefined') | join(',') }}"
+
+    - name: Verify if the machine image already exists
+      google.cloud.gcp_compute_image_info:
+        filters:
+          - name = "{{ manage_machine_image_image_name }}"
+      register: manage_machine_image__image_result
+
+    - name: Set 'manage_machine_image__image'
+      ansible.builtin.set_fact:
+        manage_machine_image__image: "{{ manage_machine_image__image | default([]) + manage_machine_image__image_result.resources }}"
+
+    - name: Print debug message
+      ansible.builtin.debug:
+        msg: A machine image {{ manage_machine_image_image_name }} already exists.
+      when: manage_machine_image__image | length == 1
+
+    - name: Delete image if exists and manage_machine_image_overwrite_image=true
+      when: manage_machine_image__image | length == 1 and manage_machine_image_overwrite_image
+      block:
+        - name: Delete machine image
+          google.cloud.gcp_compute_image:
+            zone: "{{ manage_machine_image_zone }}"
```

Comment:
This needs to have `state: absent` to delete the image. I also got an error of `Unsupported parameters for (google.cloud.gcp_compute_image) module: zone` when overwriting the image.